newrelic-scala-zio-2-api/src/test/scala/com/newrelic/zio/api/TraceOpsDSLTest.scala

```diff
@@ -0,0 +1,225 @@
+package com.newrelic.zio.api
+
+import com.newrelic.agent.introspec._
+import com.newrelic.api.agent.Trace
+import com.newrelic.zio.api.TraceOps._
+import org.junit.runner.RunWith
+import org.junit.{After, Assert, Test}
+import zio.Exit.Success
+import zio.{UIO, _}
+
+import java.util.concurrent.Executors
+import scala.concurrent.ExecutionContext
+import scala.jdk.CollectionConverters._
+
+@RunWith(classOf[InstrumentationTestRunner])
+@InstrumentationTestConfig(includePrefixes = Array("none"))
+class ZIOTraceOpsTests {
+
+  def executorService(nThreads: Int) = {
+    ExecutionContext.fromExecutor(Executors.newFixedThreadPool(nThreads))
+  }
+
+  val threadPoolOne: ExecutionContext = executorService(3)
+  val threadPoolTwo: ExecutionContext = executorService(3)
+  val threadPoolThree: ExecutionContext = executorService(3)
+
+  @After
+  def resetTxn() = {
+    com.newrelic.agent.Transaction.clearTransaction
+  }
+
+  @Test
+  def asyncTraceProducesOneSegment(): Unit = {
```

Comment:
All 6 unit tests seem to be failing.
https://github.com/newrelic/newrelic-java-agent/actions/runs/5079780519/jobs/9128030914?pr=1144

Reply:
I believe I have this resolved. Running tests now. These need to be run in a separate workflow as functional tests.

Reply:
@duartesaraiva98 I was able to resolve the testing issues.
My changes are here:
https://github.com/newrelic/newrelic-java-agent/pull/1144/files

However, testing the instrumentation locally with restful webservices in:
https://github.com/zio/zio-quickstarts
 I can see the module loads, but we lose the Transaction context and only report `NettyDispatcher` and don't see the underlying traces for the route.
 I saw this on 2.0.13; scala 2.13.8, and Java 11.